### PR TITLE
define naming and best practices for kustomize manifests

### LINKED ISSUE #48: define naming and best practices for kustomize manifests
We should add a document that is linked from README.md that defines 
- [ ] what the kustomization file should look like
- [ ] what each section does including
  - [ ] commonLabels
  - [ ] resources
- [ ] when overlays are used 
- [ ] how multiple overlays are used and their constraints
- [ ] naming of resources, groupings of resources if preferred

### PR DESCRIPTION
**Pull request by kkasravi**
fixes #48

- [x] what the kustomization file should look like
- [x] what each section does including
  - [x] commonLabels
  - [x] resources
- [x] when overlays are used 
- [x] how multiple overlays are used and their constraints
- [x] naming of resources, groupings of resources if preferred

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/56)

### CODE REVIEW

**Comment by kkasravi** [comment]
/assign @zabbasi 

please review

**Comment by kkasravi** [comment]
/assign @kunmingg 

please review and provide input

**Comment by kkasravi** [comment]
/assign @lluunn 

please review and provide input

File: docs/KustomizeBestPractices.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  These resources should be moved to a new directory under `manifests/<component>/base`. 
+
+  If moving from ksonnet, the easiest way to capture the collection of resources is to:
+
+  > use `kfctl init <appName>` to create a kubeflow application <br/>
+  > cd `<appName>`<br/>
+  > make sure the ksonnet prototype is listed in the app.yaml under components created from `kfctl init`<br/>
+  > run `kfctl generate all` to install the ksonnet packages and generate the ksonnet components <br/>
+  > cd <application>/ksonnet<br/>
+  > run `ks show <component> | tee resources.yaml`<br/>
```
Comment (gabrielwen) [inline docs/KustomizeBestPractices.md]: I think it's `ks show default/namespace -c <component>`?

File: README.md [hunk 2]
```diff
@@ -259,7 +175,7 @@ Outputs from kfctl (no platform specified):
                         ⎹→argo.yaml
                         ⎹→centraldashboard.yaml
                         ⎹→jupyter-web-app.yaml
-                        ⎹→katib.yaml
+                        →katib.yaml
```
Comment (gabrielwen) [inline README.md]: wrong format?

**Comment by kkasravi** [comment]
@jlewi - should be better. Opened #75 as well

**Pull request merged**